### ISSUE TITLE
Getting Combined response form two routes when using router.all

### ISSUE THREAD

**Issue by satishbabariya** [IBM-Swift/Kitura#1407]
## Context and Description
A description of the issue.

### Expected vs. Actual Behaviour
On calling Hello route **http://localhost:8080/hello** 

Actual response : {"hello":"world"}{"All":"Well"}
Expected response : {"hello":"world"}

Code :
```swift
import Kitura

let router = Router()

router.get("/hello") {
    request, response, next in
    response.send(json: ["hello":"world"])
    next()
}

router.all {
    request, response, next in
    response.send(json: ["All":"Well"])
    next()
}

Kitura.addHTTPServer(onPort: 8080, with: router)
Kitura.run()
```

![screen shot 2019-02-12 at 5 11 55 pm](https://user-images.githubusercontent.com/16605023/52633109-68adf780-2ee9-11e9-9b68-0c0d7d6e2f66.png)

**Comment by ddunn2** [comment]
Hi @satishbabariya 
Thanks for raising an issue. 

What is it you are trying to do exactly? And why do you think that the expected response should be: `{"hello":"world"}`? 

I have been able to replicate your issue, and if you want to prevent the `router.all` handler being executed on the `router.get("/hello")` you can modify your `router.get("/hello")` to look like this: 

```swift
router.get("/hello") {
   request, response, next in
   response.send(json: ["hello":"world"])
   try response.end()
}
```
`response.end()` ends the response and prevents the execution of any other handlers registered on `get("/hello")`. As `response.end()` does end the response, you should ensure it is called in the last handler you wish to be executed, for example if you had two routes registered like this: 

```swift
router.get("/hello") {
   request, response, next in
   response.send(json: ["hello":"world"])
   next()
}

router.get("/hello") {
   request, response, next in
   response.send(json: ["another":"hello world"])
   try response.end()
}
```
You would call `response.end()` in the last handler you wanted to be executed. 
The `next()` is used to execute the next handler or middleware registered for this `"/hello"` endpoint. 

So in summary, `response.end()` when you know you do not want any other handlers to be executed a given endpoint and `next()` when you do want to execute another handler. 

Hope this helps!

**Comment by satishbabariya** [comment]
Got it, Thanks.
